tests/unit/test_rest_streaming.py

```diff
@@ -0,0 +1,160 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import random
+from typing import List
+from unittest.mock import patch
+
+import proto
+import pytest
+import requests
+
+from google.api_core import rest_streaming
+
+
+class Composer(proto.Message):
+    given_name = proto.Field(proto.STRING, number=1)
+    family_name = proto.Field(proto.STRING, number=2)
+
+
+class Song(proto.Message):
+    composer = proto.Field(Composer, number=1)
+    title = proto.Field(proto.STRING, number=2)
+    lyrics = proto.Field(proto.STRING, number=3)
+    year = proto.Field(proto.INT32, number=4)
+
+
+class EchoResponse(proto.Message):
+    content = proto.Field(proto.STRING, number=1)
+
+
+class ResponseMock(requests.Response):
+    class _ResponseItr:
+        def __init__(
+            self,
+            _response_bytes: bytes,
+            decode_unicode: bool,
+            random_split=False,
+            seed=0,
+        ):
+            self._responses_bytes = _response_bytes
+            self._i = 0
+            self._decode_unicode = decode_unicode
+            self._random_split = random_split
+            random.seed(seed)
+
+        def __next__(self):
+            if self._i == len(self._responses_bytes):
+                raise StopIteration
+            if self._random_split:
+                n = random.randint(1, len(self._responses_bytes[self._i :]))
+            else:
+                n = 1
+            x = self._responses_bytes[self._i : self._i + n]
+            self._i += n
+            if self._decode_unicode:
+                x = x.decode("utf-8")
+            return x
+
+        def __iter__(self):
+            return self
+
+    def __init__(
+        self,
+        *args,
+        responses: List[proto.Message],
+        random_split=False,
+        response_cls,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self._responses = responses
+        self._random_split = random_split
+        self._response_message_cls = response_cls
+
+    def _parse_responses(self, responses: List[proto.Message]) -> bytes:
+        ret_val = "["
+        # json.dumps returns a string surrounded with quotes that need to be stripped
+        # in order to be an actual JSON.
+        json_responses = [
+            json.dumps(self._response_message_cls.to_json(r))[1:-1] for r in responses
+        ]
+        for x in json_responses:
+            ret_val += x
+            ret_val += ","
+        ret_val = ret_val[:-1]  # Remove trailing comma.
+        ret_val += "]"
+        return bytes(ret_val, "utf-8")
+
+    def close(self):
+        raise NotImplementedError()
+
+    def iter_content(self, *args, decode_unicode=True, **kwargs):
+        return self._ResponseItr(
+            self._parse_responses(self._responses),
+            decode_unicode,
+            random_split=self._random_split,
+        )
+
+
```

Comment:
I think that we should parametrize all the tests on `random_split` for increased coverage.

```python
pytest.mark.parametrize("random_split", [True, False]
def test_next_simple(random_split):
```

Reply:
Done